SofaKernel/modules/SofaDefaultType/src/sofa/defaulttype/DataTypeInfoRegistry.cpp

```diff
@@ -0,0 +1,60 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <sofa/defaulttype/DataTypeInfoRegistry.h>
+#include <typeindex>
+#include <map>
+#include <iostream>
+#include <sofa/defaulttype/AbstractTypeInfo.h>
+namespace sofa::defaulttype
+{
+
+std::map<const std::type_index, const AbstractTypeInfo*> typeinfos;
+
+const AbstractTypeInfo* DataTypeInfoRegistry::Get(const std::type_info& id)
+{
+    auto index = std::type_index(id);
+    std::cout << " GET GET REGISTER... "  << id.name() << "=>" << std::endl;
+    if( typeinfos.find(index) != typeinfos.end() )
+        return typeinfos[index];
+    return nullptr;
+}
+
+int DataTypeInfoRegistry::RegisterTypeInfo(const std::type_info& id, AbstractTypeInfo* info)
```

Comment:
I think it could be clearer to call it simply "set" since the class name is already quite explicit:
```
DataTypeInfoRegistry::Set(id, info);
DataTypeInfoRegistry::Get(id);
```